src/overlays/actors/ovl_En_Md/z_en_md.c

```diff
@@ -21,79 +57,813 @@ const ActorInit En_Md_InitVars = {
+        globalCtx->msgCtx.unk_E3EE = 4;
+    } else if (globalCtx->msgCtx.unk_E3EE == 3) {
+        Audio_PlaySoundGeneral(NA_SE_SY_CORRECT_CHIME, &D_801333D4, 4, &D_801333E0, &D_801333E0, &D_801333E8);
+        this->actor.textId = 0x1067;
+        func_8002F2CC(&this->actor, globalCtx, this->collider.dim.radius + 30.0f);
+
+        this->actionFunc = func_80AAB948;
+        globalCtx->msgCtx.unk_E3EE = 4;
+    } else {
+        player->stateFlags2 |= 0x800000;
+    }
+}
+
+void func_80AABD0C(EnMd* this, GlobalContext* globalCtx) {
+    func_80034F54(globalCtx, &this->unk_214, &this->unk_236, 17);
+    func_80AAA93C(this);
+
+    if (!(func_80AAB370(this, globalCtx)) || (this->unk_212 != 0)) {
+        this->actor.shape.rot = this->actor.posRot.rot;
+        return;
+    }
+
+    if (CHECK_QUEST_ITEM(QUEST_KOKIRI_EMERALD) && !(gSaveContext.eventChkInf[1] & 0x1000) && (globalCtx->sceneNum == SCENE_SPOT04)) {
+        func_80106CCC(globalCtx);
+        gSaveContext.eventChkInf[1] |= 0x1000;
+        Actor_Kill(&this->actor);
+        return;
+    }
+
+    func_80AAA92C(this, 11);
+
+    this->skelAnime.animPlaybackSpeed = 0.0f;
+    this->actor.speedXZ = 0.0f;
+    this->actor.initPosRot.pos = this->actor.posRot.pos;
+    this->actionFunc = func_80AAB8F8;
+}
+
+void EnMd_Update(Actor* thisx, GlobalContext* globalCtx) {
+    EnMd* this = THIS;
+    s32 pad;
+
+    Collider_CylinderUpdate(&this->actor, &this->collider);
+    CollisionCheck_SetOC(globalCtx, &globalCtx->colChkCtx, &this->collider);
+    SkelAnime_FrameUpdateMatrix(&this->skelAnime);
+    EnMd_UpdateEyes(&this->actor);
+    func_80AAB5A4(&this->actor, globalCtx);
+    Actor_MoveForward(&this->actor);
+    func_80AAB158(&this->actor, globalCtx);
+    func_8002E4B4(globalCtx, &this->actor, 0.0f, 0.0f, 0.0f, 4);
+    this->actionFunc(&this->actor, globalCtx);
+}
+
+s32 func_80AABEF0(struct GlobalContext* globalCtx, s32 limbIndex, Gfx** dList, Vec3f* pos, Vec3s* rot,
+                  struct Actor* thisx, Gfx** gfx) {
+    EnMd* this = THIS;
+    Vec3s vec;
+
+    if (limbIndex == 16) {
+        Matrix_Translate(1200.0f, 0.0f, 0.0f, MTXMODE_APPLY);
+        vec = this->unk_1E0.unk_08;
+        Matrix_RotateX((vec.y / 32768.0f) * M_PI, MTXMODE_APPLY);
+        Matrix_RotateZ((vec.x / 32768.0f) * M_PI, MTXMODE_APPLY);
+        Matrix_Translate(-1200.0f, 0.0f, 0.0f, MTXMODE_APPLY);
+    }
+    if (limbIndex == 9) {
+        vec = this->unk_1E0.unk_0E;
+        Matrix_RotateX((vec.x / 32768.0f) * M_PI, MTXMODE_APPLY);
+        Matrix_RotateY((vec.y / 32768.0f) * M_PI, MTXMODE_APPLY);
+    }
+
+    if (((limbIndex == 9) || (limbIndex == 10)) || (limbIndex == 13)) {
+        rot->y += Math_Sins(this->unk_214[limbIndex]) * 200.0f;
+        rot->z += Math_Coss(this->unk_236[limbIndex]) * 200.0f;
+    }
+
+    return 0;
+}
+
+void func_80AAC104(GlobalContext* globalCtx, s32 limbIndex, Gfx** dList, Vec3s* rot, Actor* actor, Gfx** gfx) {
+    s32 pad;
+    Vec3f vec = D_80AAC498;
```

Comment:
The vector data can be inlined here, which will place it in .data and do a copy at the start of the function like you want.
```suggestion
    Vec3f vec = { 400.0f, 0.0f, 0.0f };
```